Avoid an occasional null deref under Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures

whsieh opened the pull request:
#### 53f7b4f3ed1cdc390e16671f5c6e8213ad9c6136
<pre>
Avoid an occasional null deref under Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures
<a href="https://bugs.webkit.org/show_bug.cgi?id=263914">https://bugs.webkit.org/show_bug.cgi?id=263914</a>
rdar://117671779

Reviewed by Aditya Keerthi.

Add a null check and bail in `advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures`, in
the case where the `CodeBlock`&apos;s owner executable is null. Currently, we call `source()` to grab
the corresponding `JSC::SourceCode`, which assumes that `m_ownerExecutable` exists (which isn&apos;t
always the case).

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures const):

Canonical link: <a href="https://commits.webkit.org/269984@main">https://commits.webkit.org/269984@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/de7a331156c1ebbf6317f8149076b282093cc356

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24162 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2272 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25250 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26298 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22256 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24433 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3897 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24639 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/22720 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24407 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20884 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26890 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/1549 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21803 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28034 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22025 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22096 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25811 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1488 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19155 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1636 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5792 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1894 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1846 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

sysrqb reviewed (approved):
Good catch! Sorry for missing that case.

whsieh committed:
Avoid an occasional null deref under Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures
https://bugs.webkit.org/show_bug.cgi?id=263914
rdar://117671779

Reviewed by Aditya Keerthi.

Add a null check and bail in `advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures`, in
the case where the `CodeBlock`'s owner executable is null. Currently, we call `source()` to grab
the corresponding `JSC::SourceCode`, which assumes that `m_ownerExecutable` exists (which isn't
always the case).

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures const):

Canonical link: https://commits.webkit.org/269984@main
